test/periodic.jl

```diff
@@ -0,0 +1,332 @@
+@testset "periodic" begin
+    # Constructors
+    @test Periodic(sin, 5) == Periodic(sin, 5, 0)
```

Comment:
maybe test trying to construct a `Periodic(sin, 1, Day(1))` and that the fields are correctly assigned?
```julia
p.period == 5
p.phase_shift == 0
```